src/FontManagerWindows.cc

```diff
@@ -161,13 +164,17 @@ long resultFromFont(FontDescriptor **res, IDWriteFont *font) {
   return 0;
 }
 
-long getAvailableFonts(ResultSet **resultSet) {
+FontManagerImpl::FontManagerImpl() : instance_data(nullptr) {}
+FontManagerImpl::~FontManagerImpl() {}
+
+long FontManagerImpl::getAvailableFonts(ResultSet **resultSet) {
   int count = 0;
 
   IDWriteFactory *factory = NULL;
 
+  // TODO: I don't know if the "internal state" being referred to here is in the factory object or in static memory. isolate until I figure it out
   RETURN_ERROR_CODE(DWriteCreateFactory(
-    DWRITE_FACTORY_TYPE_SHARED,
+    DWRITE_FACTORY_TYPE_ISOLATED,
```

Comment:
I'm also not sure if this is needed or not. I'm not an expert in COM, so I can only comment about what I've read:
  - https://docs.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-dwritecreatefactory
      It reccommends using SHARED for better performance, but they mention ISOLATED is safer.

  - https://github.com/Axosoft/font-scanner#substitutefontpostscriptname-text it says:
    > NOTE: This seems to be producing unstable results on Windows at the moment when the test suite is run. Be cautious.

      Not sure if this could be related to leveraging the SHARED option.

  - Probably for GKC there wouldn't be a problem, since only `getAvailableFonts` is called (assuming it only reads the fonts and that the call is thread-safe), but can't tell about other applications using this add-on.

  - I've tested on Windows with both options: with SHARED the memory reference of objects `factory` and `collection` do not change between different calls. With ISOLATED they do, so at least this is working as expected.

Reply:
Yeah I'm kinda lost in this area. I think isolating the objects is probably the way to go for now just in case since performance isn't _too_ important in this case.

I think, given that we don't have any experts on this subject, we ship it as is and handle any threading issues as they popup.

Reply:
Bonus info on this subject, keeping the factories shared actually causes the windows CI to fail in interesting ways. executing the same async function twice will actually cause them to sync their execution and possibly duplicate data since the same factory is shared between both executions.

Reply:
reason enough to keep it as ISOLATED 👍